mentat/parsers/block_parser.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any, AsyncGenerator
+
+from typing_extensions import override
+
+from mentat.code_file_manager import CodeFileManager
+from mentat.config_manager import ConfigManager
+from mentat.parsers.file_edit import FileEdit
+from mentat.parsers.original_format.original_format_parsing import (
+    stream_and_parse_llm_response,
+)
+from mentat.parsers.parser import Parser
+from mentat.prompts import block_parser_prompt
```

Comment:
I'd define the string here instead of importing. Or do we plan to keep all the prompts in the same place? I guess I could see that being advantageous if some of us become more specialized "prompt engineers".

Similarly I don't see the value of importing `original_format_parsing` code instead of deleting that file and moving its code here.

Reply:
Hmm, I personally like having the prompt in a separate file; in fact, I would prefer to eventually just have it be a text file (instead of messing around with all of this python dedent stuff and strip for the prompt), and there would be a different text file for each prompt. Definitely open to suggestions on that too!

As for the original_format_parsing (and original_format_change) files, we could definitely move their code to this file; my reasoning for leaving it in those 2 files was that it's more like 'legacy' code; ideally, we don't have to touch that streaming/parsing code, and future streaming parsers we build around the current system will be cleaner (since they were built for this system). I definitely don't have anything against moving both files code to this file though!

Reply:
I could definitely see putting prompts in plain text. Though the ergonomics of importing a string are maybe a little better than reading a file.

I'm fine with leaving it separated and not cleaning it up to fit more organically in the current system. I imagine after a period of experimentation we'll settle on just one parser that we like best and delete the others from the code base. I could be wrong though. Possibly different models/languages or types of requests will work better with certain formats.

Reply:
I like having the prompts in other files as well. Right now the imported `block_parser_prompt` is our entire original system prompt. Before adding a second output format we'll have to split that into parts: 1) a generic part that explains that the model is part of a code editing system, that it might help the user write code or answer questions, etc. and 2) the output format specific part that gives a description of the response format and examples.

When we make that split lets think about the file format and explore the options @waydegg suggested

Reply:
> my reasoning for leaving it in those 2 files was that it's more like 'legacy' code; ideally, we don't have to touch that streaming/parsing code, and future streaming parsers we build around the current system will be cleaner

I'm ok leaving that code there since our intention is we find better output formats and then delete those files. I wouldn't want it hanging around as legacy code for very long though - if we find we still want this output format for something, we'd want to clean it up / integrate it more.